BaseStation/BaseStation/StartBaseStation.cs

```diff
@@ -20,17 +21,35 @@ public static class StartBaseStation
 		[STAThread]
         public static void Main(String[] args)
         {
-            //temporary example code
-            Server.BaseServer.Start(GamepadFactory.GetDriveGamePad());
-            Thread eventloop = new Thread(Server.BaseServer.EventLoop);
-            eventloop.IsBackground = true;
-            eventloop.Start();
-            Thread ipCameraControl = new Thread(() => CameraControl.Start(GamepadFactory.GetDriveGamePad()));
-            ipCameraControl.IsBackground = true;
-            ipCameraControl.Start();
+            BaseServer.Setup();
+            CameraControl.Setup();
+            new Thread(StartUpdates).Start();
             Application app = new Application();
-            app.Exit += (sd, ev) => Server.BaseServer.Shutdown();
+            app.Exit += (sd, ev) => StopUpdates();
 			app.Run(new MainWindow());
         }
+
+        private static void Update()
+        {
+            BaseServer.Update(Controller);
+            CameraControl.Update(Controller);
+            Thread.Sleep(1);
+        }
+
+        private static volatile bool exit;
+
+        public static void StartUpdates()
+        {
+            while(!exit)
+            {
+                Update();
```

Comment:
You could easily combine StartUpdates() and Update() methods because Update() isn't a large method.